lc-hax/Scripts/Patches/Dependencies/EnemyDependencyPatch.cs

```diff
@@ -2,29 +2,47 @@
 
 using HarmonyLib;
 using Hax;
+using Unity.Netcode;
 
 [HarmonyPatch(typeof(EnemyAI))]
-class EnemyDependencyPatch {
+internal class EnemyDependencyPatch {
+
     [HarmonyPatch(nameof(EnemyAI.OnDestroy))]
     static void Prefix(EnemyAI __instance) {
         _ = Helper.Enemies.Remove(__instance);
     }
 
-    [HarmonyPatch(nameof(EnemyAI.Start))]
+    [HarmonyPatch(nameof(EnemyAI.Update))]
```

Comment:
This is pretty expensive.

Reply:
> This is pretty expensive.

unless you want i couldappend a listener component thru EnemyAI and when it actives it listens to it, is easy to append a new Component listener, no need for harmony, just append them to The level enemy prefabs at start, once one spawns that component gets loaded and it will act as a listener, no hooking required.

Reply:
That's pretty bad as well, since we will be creating a game object with its own life-cycle for every enemy.

Reply:
No, no new gameobject creation as we will append directly on the prefab itself, which once spawns we need to use only Awake, we can obtain these with the dictionary i implemented here, and edit them in memory.

Reply:
I don't see how this will overcome our issue? I am probably misunderstanding what you are saying. Can you show a small example snippet?

Reply:
> I don't see how this will overcome our issue? I am probably misunderstanding what you are saying. Can you show a small example snippet?

oh  the scene on load issue? for that once you inject the mod once the game loads, which you technically are blind , as everything has already executed, for that there's a workaround, detect injection and run a few filler methods to verify and fill the blanks once.

Reply:
We can probably run some method on the static `Load` method.

Reply:
We need to simply detect if a game is loaded and active, and go from there to fill the blank lists.

Reply:
```cs
using UnityEngine;
using Hax;

public sealed class RefreshMod : MonoBehaviour {
    void Awake() {
        Helper.Enemies = Helper.FindObject<EnemyAI>();
        Helper.Grabbables = Helper.FindObject<GrabbableObject>();
        Destroy(this.gameObject);
    }
}
```

Maybe something like this?

Reply:
detect if a level is loaded first

Reply:
That's what this does

```cs
if (Helper.LocalPlayer is null) return;
```

Reply:
that could work

Reply:
Maybe this instead?

```cs
if (Helper.StartOfRound?.shipHasLanded is false) return;
```

Reply:
But this wouldn't cover if the ship is still mid-flight.

Reply:
Maybe this?

```cs
if (Helper.StartOfRound?.shipHasLanded is false && Helper.LocalPlayer?.isInElevator is false) return;
```

Reply:
yes, that can work.

Reply:
@winstxnhdw just implemented RefreshMod per your request.